src/bson.ts

```diff
@@ -112,7 +112,7 @@ export interface Document {
 const MAXSIZE = 1024 * 1024 * 17;
 
 // Current Internal Temporary Serialization Buffer
-let buffer = Buffer.alloc(MAXSIZE);
+let buffer = Buffer.allocUnsafe(MAXSIZE);
```

Comment:
Is this change intentional?  It doesn't seem to be related to adding new type errors to the Decimal128 constructor